keras_nlp/layers/modeling/rotary_embedding.py

```diff
@@ -115,8 +117,12 @@ def _compute_cos_sin_embedding(self, inputs, start_index=0):
         rotary_dim = ops.shape(inputs)[feature_axis]
         inverse_freq = self._get_inverse_freq(rotary_dim)
 
-        seq_len = ops.shape(inputs)[sequence_axis]
-        tensor = ops.arange(seq_len, dtype="float32") + start_index
+        if positions is not None:
+            # Use `positions` array instead of `start_index`
+            tensor = ops.cast(positions, dtype="float32")
```

Comment:
can we call this `positions` not `tensor`? `tensor` is the worst name.

Reply:
Agreed and done.